distuv/distribution_test.go

```diff
@@ -10,134 +10,213 @@ import (
 	"testing"
 
 	"github.com/gonum/floats"
+	"github.com/gonum/integrate/quad"
 	"github.com/gonum/stat"
 )
 
-// fullDist is a distribution that implements all the basic functions.
-type fullDist interface {
-	CDF(x float64) float64
+type meaner interface {
+	Mean() float64
+}
+
+type quantiler interface {
+	Quantile(float64) float64
+}
+
+type medianer interface {
+	quantiler
+	Median() float64
+}
+
+type varStder interface {
+	StdDev() float64
+	Variance() float64
+}
+
+type entropyer interface {
+	LogProber
 	Entropy() float64
+}
+
+type exKurtosiser interface {
 	ExKurtosis() float64
-	LogProb(x float64) float64
 	Mean() float64
-	Median() float64
-	Prob(x float64) float64
-	Quantile(p float64) float64
-	Rand() float64
-	Skewness() float64
+}
+
+type skewnesser interface {
 	StdDev() float64
+	Mean() float64
+	Skewness() float64
+}
+
+type cumulanter interface {
+	Quantiler
+	CDF(x float64) float64
 	Survival(x float64) float64
-	Variance() float64
 }
 
-// testFullDist tests all of the functions of a fullDist.
-func testFullDist(t *testing.T, f fullDist, i int) {
-	tol := 1e-1
-	const n = 1e6
-	xs := make([]float64, n)
-	for i := range xs {
-		xs[i] = f.Rand()
-	}
-	sortedXs := make([]float64, n)
-	copy(sortedXs, xs)
-	sort.Float64s(sortedXs)
-	tmp := make([]float64, n)
-
-	// Mean check.
-	mean := stat.Mean(xs, nil)
-	if !floats.EqualWithinAbsOrRel(mean, f.Mean(), tol, tol) {
-		t.Errorf("Mean mismatch case %v: want: %v, got: %v", i, mean, f.Mean())
-	} else {
-		mean = f.Mean()
+func generateSamples(x []float64, r Rander) {
+	for i := range x {
+		x[i] = r.Rand()
 	}
+}
 
-	// Median check.
-	median := stat.Quantile(0.5, stat.Empirical, sortedXs, nil)
-	if !floats.EqualWithinAbsOrRel(median, f.Median(), tol, tol) {
-		t.Errorf("Median mismatch case %v: want: %v, got: %v", i, median, f.Median())
+type probLogprober interface {
+	Prob(x float64) float64
+	LogProb(x float64) float64
+}
+
+func checkMean(t *testing.T, i int, x []float64, m meaner, tol float64) {
+	mean := m.Mean()
+	if !floats.EqualWithinAbsOrRel(mean, m.Mean(), tol, tol) {
```

Comment:
Late to the party, sorry. This effectively compares m.Mean() to m.Mean(), and x is not used.